src/main/scala/com/microsoft/hyperspace/util/ResolverUtils.scala

```diff
@@ -0,0 +1,137 @@
+ * Copyright (2020) The Hyperspace Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.microsoft.hyperspace.util
+
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.analysis.Resolver
+
+/**
+ * [[ResolverUtils]] provides utility functions to
+ * - format the index name as needed.
+ * - resolve strings based on spark's resolver.
+ */
+object ResolverUtils {
+
+  /**
+   * Normalize the index name by trimming space(s) at both ends, and replacing internal space(s)
+   * with an underscore.
+   *
+   * @param indexName the name of index passed in by user.
+   * @return a normalized index name.
+   */
+  def normalizeIndexName(indexName: String): String = {
+    indexName.trim.replaceAll("\\s+", "_")
+  }
+
+  /**
+   * Resolve two strings based on spark's resolver. Returns true if both are resolved.
+   *
+   * @param spark Spark session.
+   * @param requiredString First string.
+   * @param availableString Second string.
+   * @return True if both are equivalent based on the current spark resolver.
+   */
+  def isResolved(
+      spark: SparkSession,
+      requiredString: String,
+      availableString: String): Boolean = {
+    val resolver: Resolver = spark.sessionState.conf.resolver
+    resolver(requiredString, availableString)
+  }
+
+  /**
+   * Resolve first string from available collection of strings based on spark's resolver. Returns
+   * true if any resolved string is found in the collection.
+   *
+   * @param spark Spark session.
+   * @param requiredString String to resolve.
+   * @param availableStrings Available strings.
+   * @return True if any of the available strings can be resolved with the passed string.
+   */
+  def isResolved(
+      spark: SparkSession,
+      requiredString: String,
+      availableStrings: Iterable[String]): Boolean = {
+    availableStrings.exists(isResolved(spark, requiredString, _))
+  }
+
+  /**
+   * Resolves ALL of the requiredStrings with availableStrings based on spark's resolver. Returns
+   * true if all the requiredStrings can be resolved with some string from availableStrings.
+   * Returns false otherwise.
+   *
+   * @param spark Spark session.
+   * @param requiredStrings List of strings to resolve.
+   * @param availableStrings List of availble strings to resolve from.
+   * @return True if every one of requiredStrings can be resolved from availableStrings. False
+   *         otherwise.
+   */
+  def isResolved(
```

Comment:
how about `canResolve`? or just unify the API so that people can just use `resolve(...).isDefined`? (after addressing the different behavior below)

Reply:
done, thanks @imback82